FeedSX/screens/UniversalFeed/index.tsx

```diff
@@ -109,23 +129,83 @@ const UniversalFeed = () => {
     getInitialData();
   }, [navigationRef, lmFeedClient]);
 
-  // todo: dummy data
-  // readonly props consumed by UI component
-  const props: PostStateProps = {
-    showBookMarkIcon: true,
-    showShareIcon: true,
-    footerTextStyle: {},
-    showLabel: true,
-    labelType: 'Admin',
-  };
-
   // this calls the getFeed api whenever the page number gets changed
   useEffect(() => {
     if (accessToken) {
       fetchFeed();
     }
   }, [accessToken, feedPageNumber]);
 
+  // this function closes the post action list modal
+  const closePostActionListModal = () => {
+    setShowActionListModal(false);
+  };
+
+  // this function handles the functionality on the pin option
+  const handlePinPost = async (id: string) => {
+    let payload = {
+      postId: id,
+    };
+    dispatch(pinPostStateHandler(payload.postId) as any);
+    let pinPostResponse = await dispatch(
+      pinPost(
+        PinPostRequest.builder().setpostId(payload.postId).build(),
+      ) as any,
+    );
+    return pinPostResponse;
+  };
+
+  // this function handles the functionality on the report option
+  const handleReportPost = async () => {
+    setShowReportModal(true);
+  };
+
+  // this function handles the functionality on the delete option
+  const handleDeletePost = async (visible: boolean) => {
+    setDeleteModal(visible);
+  };
+
+  // this function returns the id of the item selected from menu list and handles further functionalities accordingly
+  const onMenuItemSelect = (postId: string, itemId?: number) => {
+    setSelectedMenuItemPostId(postId);
+    if (itemId === PIN_POST_MENU_ITEM || itemId === UNPIN_POST_MENU_ITEM) {
+      handlePinPost(postId);
+    }
+    if (itemId === REPORT_POST_MENU_ITEM) {
+      handleReportPost();
+    }
+    if (itemId === DELETE_POST_MENU_ITEM) {
+      handleDeletePost(true);
+    }
+  };
+
+  const getPostDetail = () => {
+    const postDetail = feedData.find(
+      (item: any) => item.Id === selectedMenuItemPostId,
+    );
+    return postDetail;
+  };
+   // this function invoke the share options for sharing the post link
+   const onShare = async () => {
+    try {
+      const result = await Share.share({
+        // todo: static data (replace with the deeplink) 
+        message: 'www.google.com',
+      });
+      if (result.action === Share.sharedAction) {
+        if (result.activityType) {
+          // shared with activity type of result.activityType
+        } else {
+          // shared
+        }
+      } else if (result.action === Share.dismissedAction) {
+        // dismissed
+      }
+    } catch (error: any) {
+      Alert.alert(error.message);
+    }
+  };
```

Comment:
move this code to a new file as `ShareUtils` as Share function will be total customer dependent and it should be properly readable.